src/battle_util.c

```diff
@@ -8542,7 +8659,7 @@ static inline u32 CalcMoveBasePower(u32 move, u32 battlerAtk, u32 battlerDef, u3
     switch (gBattleMoves[move].effect)
     {
     case EFFECT_PLEDGE:
-        // todo
```

Comment:
Does your damage calc code support special logic for the AI? Because right now the AI would still see a combined pledge move as two individual attacks with the regular base power. This isn't necessarily something that needs to be handled now -- you could open an issue to fix it later.

Reply:
Yeah, it does. Both damage calcs are done here, before the attack and during but you are correct with how I've done it now it would only do it during the attack.

Would something like this solve the issue? 
```
if ((move == MOVE_WATER_PLEDGE && HasMove(BATTLE_PARTNER(battlerAtk), MOVE_FIRE_PLEDGE))
 || (move == MOVE_FIRE_PLEDGE && HasMove(BATTLE_PARTNER(battlerAtk), MOVE_WATER_PLEDGE))
 || (move == MOVE_FIRE_PLEDGE && HasMove(BATTLE_PARTNER(battlerAtk), MOVE_GRASS_PLEDGE))
 || (move == MOVE_GRASS_PLEDGE && HasMove(BATTLE_PARTNER(battlerAtk), MOVE_FIRE_PLEDGE))
 || (move == MOVE_GRASS_PLEDGE && HasMove(BATTLE_PARTNER(battlerAtk), MOVE_WATER_PLEDGE))
 || (move == MOVE_WATER_PLEDGE && HasMove(BATTLE_PARTNER(battlerAtk), MOVE_GRASS_PLEDGE)))
    basePower = 150;
```
This does seem like a lot of checks but I'm not sure if there is a cleaner way.

Reply:
You wouldn't want to check if the partner has the move, more so if they've selected the move. One Pokemon will always choose its attack before the other, so you can check if the partner has actually selected an attack, and if so check if it selected a pledge move different from the one currently being checked.

Reply:
I don't think this can work in the expansion because damage is calculated each turn before any moves were chosen.

Reply:
Hm, maybe there could be an exception made for the pledge effect to recalc everything after the first move was selected.

Reply:
I wasn't able to do it. Would an issue ticket be okay? Maybe someone who understands this part of the code base better could look into it.